store/sql_post_store.go

```diff
@@ -642,6 +642,7 @@ func (s SqlPostStore) Search(teamId string, userId string, params *model.SearchP
 				Posts
 			WHERE
 				DeleteAt = 0
+				AND Type NOT LIKE '` + model.POST_SYSTEM_MESSAGE_PREFIX + `%'
```

Comment:
It's not the length, it's the wildcard LIKE that's going to cause a table scan (I think).  I think if you look at the query profiler it may show a table scan.  Post processing the returned results will be slower than the LIKE clause in the db.  Plus you could in theory filter the results to zero even though there are valid results because we use LIMIT.

It's probably not a huge deal since I think it will run with the full text index first then scan those vs a full table scan (I think).

For some reason I thought system messages had an ID of `0`, but I forgot we used the type field.  Since ID has an index as well it would have been more performant.

+1